### ISSUE TITLE
CLONE recursive true should return all cloned widgets as collection

### ISSUE THREAD

**Issue by Ingegneus** [ArnoldSmith86/virtualtabletop#1661]
imo when I clone all widgets with recursive:true the collection should contain all widgets that where cloned.

**Comment by robartsd** [comment]
I think there should be a new collection that contains all the widgets cloned in addition to the current collection that contains the root widget(s) cloned.

**Comment by flackr** [comment]
I think it would be better to provide a routine that gets all descendant widgets into a collection, but I'd be interested to hear what you might do with a collection of all the descendants too.

For my purposes I think it's useful to have a collection of just the root widgets.

**Comment by Ingegneus** [comment]
> I think there should be a new collection that contains all the widgets cloned in addition to the current collection that contains the root widget(s) cloned.

this seems like a good idea

> I think it would be better to provide a routine that gets all descendant widgets into a collection, but I'd be interested to hear what you might do with a collection of all the descendants too.

why not both? :)

> For my purposes I think it's useful to have a collection of just the root widgets.

for example in my implementation of smash up, I have a minimap, that shows the cards played even on regions that are off screen. for that I need to clone all the widgets that have a certain parent and apply properties to them

**Comment by 96LawDawg** [comment]
> > I think there should be a new collection that contains all the widgets cloned in addition to the current collection that contains the root widget(s) cloned.
> 
> this seems like a good idea
> 

Agree.  Two collections.  Not sure how those would be named or whatever.

> for example in my implementation of smash up, I have a minimap, that shows the cards played even on regions that are off screen. for that I need to clone all the widgets that have a certain parent and apply properties to them

If that doesn't get implemented, you could always SELECT the widgets that have the property "clonedFrom" of the widget you want and then apply those properties directly.  I know.  Your proposal is easier to implement in the JSON; just suggesting a workaround.

**Comment by flackr** [comment]
I have no objections to additional options as long as we support the base case of only wanting the root.

> for example in my implementation of smash up, I have a minimap, that shows the cards played even on regions that are off screen. for that I need to clone all the widgets that have a certain parent and apply properties to them

In this case, wouldn't you want to select everything that has a certain parent, specify those as the source for the CLONE, and then you'd get a collection of the clones of all of those widgets? Or do you want to clone the parent too and apply those properties to the parent as well?
